test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -110,34 +109,54 @@ def setUp(self):
         self.ATTR_V = 'Full_Variable_List'
         api_to_cli.setdefault(self.ATTR_V, 'V')
 
+        # temporary files
+        (fd, fn) = tempfile.mkstemp(prefix="job_out1")
+        os.close(fd)
+        self.job_out1_tempfile = fn
+
+        (fd, fn) = tempfile.mkstemp(prefix="job_out2")
+        os.close(fd)
+        self.job_out2_tempfile = fn
+
+        (fd, fn) = tempfile.mkstemp(prefix="job_out3")
+        os.close(fd)
+        self.job_out3_tempfile = fn
+
+    def tearDown(self):
+        try:
+            os.remove(self.job_out1_tempfile)
+            os.remove(self.job_out2_tempfile)
+            os.remove(self.job_out3_tempfile)
+
+        except:
+            pass
```

Comment:
Avoid using try/except pass.  You have a good idea of what exception that might be thrown.  Except that.  What happens if someone does a keyboard interrupt or there is a memory error?  These exceptions will be caught and nothing will be done about them.  I believe the exception you are looking for is OSError.  Also, move the call to the parent's tearDown() to the top of the function.

Reply:
I've updated the code to do an except OSError.
I'm confused about your second comment. The try...except is called in tearDown() function first, and at "finally" step, it goes and calls TestFunctional.tearDown(self). My intension is that whether or not the os.remove() calls fail or not on OsError, I would still call the parent tearDown().

Reply:
I've updated the code to do an except OSError.
I'm confused about your second comment. The try...except is called in tearDown() function first, and at "finally" step, it goes and calls TestFunctional.tearDown(self). My intension is that whether or not the os.remove() calls fail or not on OsError, I would still call the parent tearDown().

Reply:
Unless you have an explicit reason, it is almost always the right idea to call the parent tearDown() as the first thing.  If you call it first, it will always be called regardless of the state of the removals.

Reply:
Ok, then, I'll call parent tearDown() first, which eliminates 'finally'.